### PR TITLE
Minor logic simplification when recovering changeview

### PR DESCRIPTION
**Pull request by vncoelho**
Avoiding trying to increase change view if commitsent flag is already set

### CODE REVIEW

**Comment by vncoelho** [comment]
I think that we should also move  `isRecovering = true`; inside try.

File: neo/Consensus/ConsensusService.cs [hunk 1]
```diff
@@ -280,16 +280,17 @@ private void OnRecoveryMessageReceived(ConsensusPayload payload, RecoveryMessage
...
-                    ConsensusPayload[] changeViewPayloads = message.GetChangeViewPayloads(context, payload);
-                    foreach (ConsensusPayload changeViewPayload in changeViewPayloads)
-                        ReverifyAndProcessPayload(changeViewPayload);
-                }
-                if (message.ViewNumber != context.ViewNumber) return;
                 if (!context.CommitSent())
                 {
+                    if (message.ViewNumber > context.ViewNumber)
+                    {
+                        if (context.CommitSent()) return;
```
Comment (shargon) [inline neo/Consensus/ConsensusService.cs]: In line 283 you have the opposite verification

Comment (vncoelho) [reply neo/Consensus/ConsensusService.cs]: Yessss, sorry for that, we just removed line 283. You and Jeff suggested at the same time...aehauhea

**Comment by vncoelho** [comment]
It was correct as it is, almost no possibility for simplification, thanks @shargon and @jsolman.

**Comment by jsolman** [comment]
NO problem, always happy to discuss offline :-)

**Comment by vncoelho** [comment]
I include a brief description of the code in the comment above. Now with the yellow paper we gonna have this better explained.
The flowchart is 90% complete.

https://github.com/NeoResearch/yellowpaper/blob/master/sections/graphviz-images/graphviz-dbft-v2-recover.jpg

**Pull request closed without merging**